windows cygwin shell fails for install

**Describe the bug**
found in https://stackoverflow.com/questions/70893995/jbang-on-cygwin-shell

```
curl -Ls https://sh.jbang.dev | bash -s - app install --fresh --force quarkus@quarkusio
Downloading JBang... 
Error downloading JBang
```

I  confirm this happens using cygwin shell. not sure why yet.

I get something different:

```
$ curl -Ls https://sh.jbang.dev | bash -x -s - app install --fresh --force quarkus@quarkusio
+ javaVersion=11
++++ absolute_path bash
++++ [[ bash = /* ]]
++++ [[ bash =~ ^[A-z]:/ ]]
++++ echo /home/root/bash
+++ resolve_symlink /home/root/bash
+++ [[ cygwin != darwin* ]]
+++ minusFarg=-f
++++ readlink -f /home/root/bash
+++ sym_resolved=/home/root/bash
+++ [[ -n /home/root/bash ]]
+++ echo /home/root/bash
++ dirname /home/root/bash
+ abs_jbang_dir=/home/root
+ libc_type=glibc
+ file_type=tar.gz
+ case "$(uname -s)" in
++ uname -s
+ os=windows
+ libc_type=c_std_lib
+ file_type=zip
+ case "$(uname -m)" in
++ uname -m
+ arch=x64
+ [[ 11 -eq 8 ]]
+ [[ 11 -eq 11 ]]
+ distro=temurin
+ [[ -z '' ]]
+ JBDIR=/home/root/.jbang
+ [[ -z '' ]]
+ TDIR=/home/root/.jbang/cache
+ '[' -f /home/root/jbang.jar ']'
+ '[' -f /home/root/.jbang/jbang.jar ']'
+ [[ ! -f /home/root/.jbang/bin/jbang.jar ]]
+ [[ ! -f /home/root/.jbang/bin/jbang ]]
+ /home/root/.jbang/bin/jbang app install --fresh --force quarkus@quarkusio
Error: Could not find or load main class dev.jbang.Main
Caused by: java.lang.ClassNotFoundException: dev.jbang.Main
+ exit 1
```

Dunno what's going on, but it all seems completely broken on Cygwin.
I can fix the `curl` command but then the problem just moved to the error I pasted earlier https://github.com/jbangdev/jbang/issues/1223#issuecomment-1024412877

And I can trace it back to this weirdness:

```
~ $ java -classpath ~/.jbang/bin/jbang.jar dev.jbang.Main version
Error: Could not find or load main class dev.jbang.Main
Caused by: java.lang.ClassNotFoundException: dev.jbang.Main
~ $ java -classpath home/tako/.jbang/bin/jbang.jar dev.jbang.Main version
Error: Could not find or load main class dev.jbang.Main
Caused by: java.lang.ClassNotFoundException: dev.jbang.Main
~ $ java -classpath .jbang/bin/jbang.jar dev.jbang.Main version
0.87.0
~ $ cd .jbang/bin
~/.jbang/bin $ java -classpath ~/.jbang/bin/jbang.jar dev.jbang.Main version
Error: Could not find or load main class dev.jbang.Main
Caused by: java.lang.ClassNotFoundException: dev.jbang.Main
~/.jbang/bin $ java -classpath jbang.jar dev.jbang.Main version
0.87.0
```

So somehow it runs okay when using relative paths but breaks as soon as you use absolute paths. The same happens with the `curl` command btw. Relative paths for the `-o` parameter work, absolute paths don't. (This includes `/tmp` for examples, so it's not just `/home/....`)

--- a/src/main/scripts/jbang
+++ b/src/main/scripts/jbang
@@ -110,6 +110,12 @@ case "$(uname -m)" in
     ;;
 esac
 
+## when using cygwin fall out to just running the bat file.
+if [[ $os == windows && -f "$abs_jbang_dir/jbang.cmd" && "$(uname -s)" == CYGWIN* ]]; then
+  cmd /c "$(cygpath -m "$abs_jbang_dir"/jbang.cmd)" "$@"
+  exit $?
+fi
+
 if [[ "$javaVersion" -eq 8 || "$javaVersion" -eq 11 || "$javaVersion" -ge 17 ]]; then distro="temurin"; else distro="aoj"; fi
 
 if [[ -z "$JBANG_DIR" ]]; then JBDIR="$HOME/.jbang"; else JBDIR="$JBANG_DIR"; fi

--- a/src/main/scripts/jbang
+++ b/src/main/scripts/jbang
@@ -17,21 +17,21 @@ absolute_path() {
 
 resolve_symlink() {
   if [[ $OSTYPE != darwin* ]]; then minusFarg="-f"; fi
-  sym_resolved=$(readlink ${minusFarg} $1)
+  sym_resolved=$(readlink ${minusFarg} "$1")
 
   if [[ -n $sym_resolved ]]; then
-    echo $sym_resolved
+    echo "$sym_resolved"
   else
-    echo $1
+    echo "$1"
   fi
 }
 
 download() {
   if [ -x "$(command -v curl)" ]; then
-    curl -sLf -o "$2" $1
+    curl -sLf -o "$2" "$1"
     retval=$?
   elif [ -x "$(command -v wget)" ]; then
-    wget -q -O "$2" $1
+    wget -q -O "$2" "$1"
     retval=$?
   else
     echo "Error: curl or wget not found, please make sure one of them is installed" 1>&2
@@ -50,7 +50,7 @@ unpack() {
       mv "$TDIR/jdks/$javaVersion.tmp/"*/* "$TDIR/jdks/$javaVersion.tmp/"
     fi
   else
-    unzip -qq -o $1 -d "$2"
+    unzip -qq -o "$1" -d "$2"
     retval=$?
     mv "$TDIR/jdks/$javaVersion.tmp/"*/* "$TDIR/jdks/$javaVersion.tmp/"
   fi
@@ -60,7 +60,7 @@ javacInPath() {
   [[ -x "$(command -v javac)" && ( $os != "mac" || $(/usr/libexec/java_home &> /dev/null) ) ]]
 }
 
-abs_jbang_dir=$(dirname $(resolve_symlink $(absolute_path $0)))
+abs_jbang_dir=$(dirname "$(resolve_symlink "$(absolute_path "$0")")")
 
 # todo might vary by os so can be overwritten below
 libc_type=glibc
@@ -141,7 +141,7 @@ else
     rm -f "$JBDIR/bin/jbang" "$JBDIR/bin"/jbang.*
     cp -f "$TDIR/urls/jbang/bin"/* "$JBDIR/bin"
   fi
-  $JBDIR/bin/jbang "$@"
+  "$JBDIR"/bin/jbang "$@"
   exit $?
 fi
 if [ -f "$jarPath.new" ]; then
@@ -153,7 +153,7 @@ fi
 unset JAVA_EXEC
 if [[ -n "$JAVA_HOME" ]]; then
   # Determine if a (working) JDK is available in JAVA_HOME
-  if [ -x "$(command -v $JAVA_HOME/bin/javac)" ]; then
+  if [ -x "$(command -v "$JAVA_HOME"/bin/javac)" ]; then
     JAVA_EXEC="$JAVA_HOME/bin/java"
   else
     echo "JAVA_HOME is set but does not seem to point to a valid Java JDK" 1>&2
@@ -184,7 +184,7 @@ if [[ -z "$JAVA_EXEC" ]]; then
       mkdir -p "$TDIR/jdks"
       echo "Downloading JDK $javaVersion. Be patient, this can take several minutes..." 1>&2
       jdkurl="https://api.foojay.io/disco/v2.0/directuris?distro=$distro&javafx_bundled=false&libc_type=$libc_type&archive_type=$file_type&operating_system=$os&package_type=jdk&version=$javaVersion&release_status=ga&architecture=$arch&latest=available"
-      download $jdkurl "$TDIR/bootstrap-jdk.$file_type"
+      download "$jdkurl" "$TDIR/bootstrap-jdk.$file_type"
       if [ $retval -ne 0 ]; then echo "Error downloading JDK" 1>&2; exit $retval; fi
       echo "Installing JDK $javaVersion..." 1>&2
       rm -rf "$TDIR/jdks/$javaVersion.tmp/"
@@ -199,7 +199,7 @@ if [[ -z "$JAVA_EXEC" ]]; then
       # Activate the downloaded JDK giving it its proper name
       mv "$TDIR/jdks/$javaVersion.tmp" "$TDIR/jdks/$javaVersion"
       # Set the current JDK
-      ${JAVA_EXEC} -classpath ${jarPath} dev.jbang.Main jdk default $javaVersion
+      ${JAVA_EXEC} -classpath "${jarPath}" dev.jbang.Main jdk default $javaVersion
     fi
   fi
 fi
@@ -209,11 +209,11 @@ fi
 
 ## run it using command substitution to have just the user process once jbang is done
 export JBANG_RUNTIME_SHELL=bash
-output=$(CLICOLOR_FORCE=1 ${JAVA_EXEC} ${JBANG_JAVA_OPTIONS} -classpath ${jarPath} dev.jbang.Main "$@")
+output=$(CLICOLOR_FORCE=1 ${JAVA_EXEC} ${JBANG_JAVA_OPTIONS} -classpath "${jarPath}" dev.jbang.Main "$@")
 err=$?
 if [ $err -eq 255 ]; then
   eval "exec $output"
-elif [ ! -z "$output" ]; then
+elif [ -n "$output" ]; then
   echo "$output"
   exit $err
 else

